### PR TITLE
Separate out rai_core_flask CI and add paths filter

### PR DESCRIPTION
**Pull request by romanlutz**
## Description

With this change the rai_core_flask CI will only run if we make changes to that folder which is very rare. This should reduce the ridiculous amount of requeuing pipelines significantly.

## Checklist

- [ ] I have added screenshots above for all UI changes.
- [ ] I have added e2e tests for all UI changes.
- [ ] Documentation was updated if it was needed.

### CODE REVIEW

File: .github/workflows/CI-python.yml [hunk 1]
```diff
@@ -25,9 +24,6 @@ jobs:
           [ubuntu-20.04, ubuntu-latest, macos-latest, windows-latest]
         pythonVersion: ["3.6", "3.7", "3.8", "3.9"]
         exclude:
-          - packageDirectory: "rai_core_flask"
-            operatingSystem: macos-latest
-            pythonVersion: "3.9"
           - operatingSystem: ubuntu-20.04
```
Comment (gaugup) [inline .github/workflows/CI-python.yml:27]: Can we remove the setting of fail_fast flag as well? I don't think we should run tests that will fail anyways.

**Review by gaugup — COMMENTED** [review]
Can we remove the setting of fail_fast flag as well? I don't think we should run tests that will fail anyways.

File: .github/workflows/CI-python.yml (hunk 1 above)
Comment (romanlutz) [reply .github/workflows/CI-python.yml:27]: I suppose for this particular one we can turn it off. I would still like to keep it for the notebook e2e tests, though. Those take ages (1h+) and one of them fails semi-frequently. I'll remove it for the python one.

Comment (gaugup) [reply .github/workflows/CI-python.yml:24]: great. Thanks.

**Review by gaugup — APPROVED** [review]

**Comment by romanlutz** [comment]
This only touches the CI python pipelines which are passing. Everything else looks to be passing, too, but I'd have to rerun them now again since another PR got merged. Since I've spent way too much time on rerunning builds today (and waiting for this to merge so that at least the other 5 PRs can run their builds somewhat faster) I'll make an exception and merge this by bypassing branch protections.

**Pull request merged**